modeling/preprocessing.py

```diff
@@ -82,9 +80,26 @@ def convert_to_floats(df, descriptions):
 
     return df_modified
 
+
+
 def handle_outliers(df):
     """
     This function will detect outliers in the data. and replace them with missing values
     """
     # TODO: Implement this function
-    return df
+    df_modified = df.copy()
+    column_names = df.columns.tolist()
+    score_headers = []
+    score_headers = df.columns[df.columns.str.contains('_score')]
+    upper_quartiles = df_modified[score_headers].quantile(0.75)
+    lower_quartiles = df_modified[score_headers].quantile(0.25) 
+    iqr = upper_quartiles - lower_quartiles
+    outliers = (df_modified[score_headers] > (upper_quartiles + 1.5 * iqr)) | (df_modified[score_headers] < (lower_quartiles - 1.5 * iqr))
```

Comment:
an edge case I thought was what if the data has a lot of zeros (or missing values), we might need to fill those with NaNs to begin with.